//
// PROJECT:         Aspia
// FILE:            protocol/authorization.proto
// LICENSE:         GNU General Public License 3
// PROGRAMMERS:     Dmitry Chapyshev (dmitry@aspia.ru)
//

syntax = "proto3";

option optimize_for = LITE_RUNTIME;

package aspia.proto.auth;

enum SessionType
{
    SESSION_TYPE_UNKNOWN        = 0;
    SESSION_TYPE_DESKTOP_MANAGE = 1;
    SESSION_TYPE_DESKTOP_VIEW   = 2;
    SESSION_TYPE_FILE_TRANSFER  = 4;
}

enum Status
{
    STATUS_UNKNOWN       = 0;
    STATUS_SUCCESS       = 1;
    STATUS_ACCESS_DENIED = 2;
    STATUS_CANCELED      = 3;
}

enum Hashing
{
    HASHING_UNKNOWN = 0;
    HASHING_SHA512  = 1;
}

// Host to client.
message Request
{
    Hashing hashing  = 1;
    uint32 rounds    = 2;
    bytes nonce      = 3;
}

// Client to host.
message Response
{
    SessionType session_type = 1;
    string username = 2;
    bytes key = 3;
}

// Host to client.
message Result
{
    Status status = 1;
}
